### ISSUE TITLE
Question regarding Form ACTION and Webhook

### ISSUE THREAD

**Issue by icoso99** [pojome/elementor#4742]
In reference to this closed post: https://github.com/pojome/elementor/issues/4534

I apologize, However, I am VERY new to this and using Elementor. I am assuming that this code is to be added to the webhook.php file, right or no? IF no, then where is this php code to be added?

I've researched and read numerous post about using the Webhook feature and forms Action tag, here and on Facebook forum, and still a bit confused.

Also I found this page for Webhooks on Forms? https://docs.elementor.com/article/281-form-faq Which explains that "To control the field names, give your form fields a custom ID and use the following snippet. Be sure to update the form name and your webhook URL." Which we have done. I added the exact code from that page to the webhook.php file as follows:

```PHP
add_action( 'elementor_pro/forms/new_record', function( $record, $handler ) {
     //make sure its our form
     $form_name = $record->get_form_settings( 'form_name' );
     if ( 'loanform' !== $form_name ) {
         return;

     }

     $raw_fields = $record->get( 'fields' );
     $fields = [];
     foreach ( $raw_fields as $id => $field ) {
	$fields[ $id ] = $field['value'];
     }
     wp_remote_post( 'HTTP://hometownfcu.org/wp/loan-calc-form', [
	     'body' => $fields,
     ]);
});

```

Then in my Elementor pro form for the webhook URL I put: HTTP://myurl.org/wp/loan-calc-form
This loan-calcform is the same page that contains this form. I'm trying to send the data to this page itself. so the calculator that I built using PHP will populate the form. You can find an example of what I m try to accomplish here: http://icoso.com/loancalc/. This is a BASIC webpage. this is not an elementor page. But on this page my form tag looks like:
form METHOD="POST" ACTION="" name="loanform" id="loanform"
and my calculate button is a simple submit button:
input type="submit" value="Calculate" id="calcLn" name="calcLn"

My Php Code tests for if ($calcLn) then does the calculation with the submitted code and runs a javascript that my php code generates that populates the readonly fields at the bottom of this form page.

On the Elementor pro page I've created the form, however using the code inspector the FORM tag does not appear to have any type of ACTION tag, how do we add this to a form? I figured I'd use the Webhook and send the date to the form itself that way, ie: http://hometownfcu.org/wp/loan-calc-form/ , I dont appear to get anything submitted to the form I created in Elementor Pro. I'm using PHP Everywhere to pull in my background PHP via a Shortcode that is put into a HTML field that is loaded above my FORM. I know this is running, because I added a series of PHP echo's to display info on the page to see how far I got. I've put two type of button on my calc form, an HTML button that uses and the Elementor Submit button. I've tested with both buttons to submit the data. at the bottom of the form page the words "The form was sent successfully" gets added to the page.

I also found this: https://developers.elementor.com/forms-api/
On this forms API it talks about this:
Form Submission Action
elementor_pro/forms/form_submitted
After the forms module is loaded and it’s a POST request with the form action. This is the place to add a form handlers.

And it gives code examples. Where does this code go? Where do I add or edit this code?

Basically, I'm trying to create an Elementor Pro form on my Wordpress site, that acts like my basic calculator at http://icoso.com/loancalc/

Any suggestions would be greatly appreciated.

**Comment by bainternet** [comment]
@icoso99 

You made a big mess of things and this will not work for you.

First, the form is submitted via AJAX which means it needs no `action` attribute. Second, the snippet you posted is used to create a custom webhook so there is no need for you to set up the webhook URL in the form settings. And 3rd you can process your form on the action instead of creating a request.


Anyway, Is seems like you need some custom coding help more then simple guidance.
Try asking on Elementor Community facebook group: https://www.facebook.com/groups/Elementors/

**Comment by icoso99** [comment]
Thanks for the info.  I was asking for some clarification of these things as the documentation of the Webhook, and/or the forms-api is woefully inadequate in explaining where or how to use these features.  I've even read through the Elementor Developers Tutorials. Even there is nothing that thoroughly explains how or where to implement these "hooks".  There are descriptions of the hooks and how they work, but this is all high level coding instruction and assumes a lot, no examples, no descriptions, no feedback from other developers, etc. 

Regarding the Webhook snippet, the instructions on that page page state "To control the field names, give your form fields a custom ID and use the following snippet. Be sure to update the form name and your webhook URL."  I assumed that meant using the Webhook URL in the editor.  Simply adding "in the code below. " would make the documention much clearer.   There is nothing on that page that explains where to use that snippet of code and why.  After looking at all the options on that page editor, and much research, I assumed it had to be added to a file somewhere and discovered the webhook.php file. Which according to some other documentation that I found had the wrong URL for its location.  I then added it to the webhook.php file and tested with it.  It still didn't work as I expected it to. The button values seemingly were never passed back to the form itself. Or at least not as I expected them to be. Again, there is little to no documentation explaining this process thoroughly.  Ever use PHP.net? There is a great community forum there that helps on every PHP method or function and posts with examples, etc are extremely helpful.  

Thirdly, your explanation "And 3rd you can process your form on the action instead of creating a request." 
make no sense to me.  Sorry, I'm not an extremely knowledgeable coding expert, nor an expert with WordPress and how everything is processed.  I'm just starting out with WordPress actually,  but this tells me nothing nor is helpful at all.  You first state there is "no action attribute" then you say I can process on the action? 

I looked through the facebook group.  It really has no real help either.  Its a facebook page... 

Thanks for your feedback .  Elementor is a great product.

**Comment by bkjproductions** [comment]
The Facebook group seems to be "closed" and I cannot join it? Not that I really want to, but what I apparently need to do is override the normal "on submit" of the Elementor form and change it to my own script. I want to change the form ACTION to a different URL, so that the data from this form goes there--- it's a search results page that will show the results when you fill out this form.
